src/syscheckd/db/src/db.cpp

```diff
@@ -95,6 +114,60 @@ void fim_sync_push_msg(const char* msg)
     }
 }
 
+TXN_HANDLE fim_db_transaction_start(const char* table, result_callback_t row_callback, void *user_data)
+{
+    const std::unique_ptr<cJSON, CJsonDeleter> jsInput
+    {
+        cJSON_Parse(table)
+    };
+
+    callback_data_t cb_data = { .callback = row_callback, .user_data = user_data };
+
+    TXN_HANDLE dbsyncTxnHandle = dbsync_create_txn(FIMDB::getInstance().DBSyncHandle(), jsInput.get(), 0,
+                                                   QUEUE_SIZE, cb_data);
+
+    return dbsyncTxnHandle;
+}
+
+FIMDBErrorCodes fim_db_transaction_sync_row(TXN_HANDLE txn_handler, const fim_entry* entry) {
+
+    nlohmann::json json_insert;
+    auto retVal = FIMDB_OK;
+
+    if (entry->type == FIM_TYPE_FILE)
+    {
+        auto syncItem = std::make_unique<FileItem>(entry);
+        json_insert = *syncItem->toJSON();
+    }
+
+    const std::unique_ptr<cJSON, CJsonDeleter> jsInput
+    {
+        cJSON_Parse(json_insert.dump().c_str())
+    };
+
+
+    int res = dbsync_sync_txn_row(txn_handler, jsInput.get());
+    if (res != 0)
+    {
+        retVal = FIMDB_ERR;
+    }
+
+    return retVal;
+}
+
+FIMDBErrorCodes fim_db_transaction_deleted_rows(TXN_HANDLE txn_handler,
```

Comment:
If this will be call from C you don't have FIMDBErrorCodes

Reply:
It is missing exception management in these functions

Reply:
We are using the C interface of DBSync. These functions are performing the exception management and returning different error codes depending on the exception that has been triggered.